staging/src/k8s.io/code-generator/cmd/prerelease-lifecycle-gen/prerelease-lifecycle-generators/status_test.go

```diff
@@ -146,3 +147,293 @@ func Test_extractKubeVersionTag(t *testing.T) {
 		})
 	}
 }
+
+func Test_extractTag(t *testing.T) {
+	comments := []string{
+		"+variable=7",
+		"+anotherVariable=8",
+		"+yetAnotherVariable=9",
+		"variableWithoutMarker=10",
+		"+variableWithoutValue",
+		"+variable=11",
+		"+multi-valuedVariable=12,13,14",
+		"+strangeVariable=15,=16",
+	}
+
+	tests := []struct {
+		name         string
+		tagComments  []string
+		variableName string
+		wantValue    *tagValue
+	}{
+		{
+			name:         "variable with explicit value",
+			tagComments:  comments,
+			variableName: "anotherVariable",
+			wantValue:    &tagValue{value: "8"},
+		},
+		{
+			name:         "variable without explicit value",
+			tagComments:  comments,
+			variableName: "variableWithoutValue",
+			wantValue:    &tagValue{value: ""},
+		},
+		{
+			name:         "variable not present in comments",
+			tagComments:  comments,
+			variableName: "variableOutOfNowhere",
+			wantValue:    nil,
+		},
+		{
+			name:         "variable without marker test",
+			tagComments:  comments,
+			variableName: "variableWithoutMarker",
+			wantValue:    nil,
+		},
+		/* The following test cases should abort the execution */
+		/*
+			{
+				name:         "duplicated variable",
+				tagComments:  comments,
+				variableName: "variable",
+				wantValue:    nil, // TODO: Today it calls klog.Fatal, check how to capture it
+			},
+			{
+				name:         "variable without marker test",
+				tagComments:  comments,
+				variableName: "multi-valuedVariable",
+				wantValue:    nil, // TODO: Today it calls klog.Fatal, check how to capture it
```

Comment:
@pohly didn't we have some klog-testing feature to test this?

Reply:
You remember correctly. `klog.OsExit` gets exported and can be overridden:

https://github.com/kubernetes/klog/blob/6632ba5cc9a5331c853a2c6dbc0fbb125e6e7b94/exit.go#L38-L43

Reply:
Thanks @aojea and @pohly for the hint! I used it in the newly added test and also in the the pre-existing one.